hazelcast/src/main/java/com/hazelcast/core/RingbufferStore.java

```diff
@@ -26,31 +26,48 @@
     /**
      * Stores one item with it's corresponding sequence.
      *
+     * Exceptions thrown by this method causes the primary and the backups of the
+     * {@link com.hazelcast.ringbuffer.Ringbuffer} to having a different view of the Ringbuffer as the exception prevents
+     * adding the data on the backups.
+     *
      * @param sequence the sequence ID of the data
      * @param data     the value of the data to store
+     * @see com.hazelcast.ringbuffer.Ringbuffer#add(Object)
+     * @see com.hazelcast.ringbuffer.Ringbuffer#addAsync(Object, com.hazelcast.ringbuffer.OverflowPolicy)
      */
     void store(long sequence, T data);
 
     /**
-     * Stores multiple entries. Implementation of this method can optimize the store operation.
+     * Stores multiple items. Implementation of this method can optimize the store operation.
+     *
+     * Exceptions thrown by this method causes the primary and the backups of the
```

Comment:
I took a look at the map store and it seems the map operations first call the map store before changing the in-memory state. Can you change the order with the ringbuffer so we avoid this issue instead? So first call the store, then add it to the in-memory ringbuffer.